demo/odaslive/parameters.c

```diff
@@ -244,6 +257,42 @@
 
     }
 
+    pa_channel_map* parameters_pa_channel_map_config(const char* fileConfig)
+    {
+        unsigned int nChannels = parameters_count(fileConfig, "raw.interface.channelmap");
+        pa_channel_map* map = (pa_channel_map*) malloc(sizeof(pa_channel_map));
+        pa_channel_map_init(map);
+        map->channels = 0;
+
+        char * tmpStr1 = (char *) malloc(sizeof(char) * 1024);
+        for (unsigned int iChannel = 0; iChannel < nChannels; ++iChannel)
+        {
+            sprintf(tmpStr1, "raw.interface.channelmap.[%u]", iChannel);
+            char* channel = parameters_lookup_string(fileConfig, tmpStr1);
+            if (((map->map[iChannel] = pa_channel_position_from_string(channel)) == PA_CHANNEL_POSITION_INVALID))
+            {
+                printf("Invalid channel position: %s\n", channel);
+                exit(EXIT_FAILURE);
+            }
+            ++(map->channels);
+            free((void *) channel);
+        }
+        free((void *) tmpStr1);
+
+        if (map->channels != parameters_lookup_int(fileConfig, "raw.nChannels"))
+        {
+            printf("Configuration error: raw.interface.channelmap length does not match raw.nChannels\n");
+            exit(EXIT_FAILURE);
+        }
+
+        if (!pa_channel_map_valid(map))
+        {
+            printf("Invalid channel map\n");
+            exit(EXIT_FAILURE);
+        }
+        
+        return map;
+    }
```

Comment:
Removed alternative implementation for `channelnames = "front-left,front-right";` style